app/controller_services/application_controller/authorization_service.rb

```diff
@@ -1,26 +1,51 @@
 # frozen_string_literal: true
 
 require 'service/unauthorized_result'
-require 'service/internal_server_error_result'
 
 class ApplicationController < ActionController::API
   class AuthorizationService
-    def initialize(controller)
-      @controller = controller
+    FIREBASE_VERIFICATION_URI = 'https://www.googleapis.com/identitytoolkit/v3/relyingparty/getAccountInfo'
+
+    class AmbiguousUserError < StandardError; end
+
+    def initialize(controller, access_token)
+      @controller   = controller
+      @access_token = access_token
     end
 
     def perform
-      if User.first.nil?
-        Rails.logger.error 'No users exist'
-        return Service::InternalServerErrorResult.new(errors: ['Attempted to set current user but there are no users'])
-      end
+      return Service::UnauthorizedResult.new(errors: 'No Google OAuth 2.0 access token found') if access_token.blank?
+
+      token_response = connection.post {|req| req.body = { idToken: access_token }.to_json }
+
+      if token_response.success?
+        users = JSON.parse(token_response.body)['users']
 
-      controller.current_user = User.first
-      nil
+        raise AmbiguousUserError.new('Token validation response did not include a user') if users.blank?
+        raise AmbiguousUserError.new('Token validation response included multiple users') if users.length > 1
+
+        controller.current_user = User.create_or_update_for_google(users.first)
+        nil
+      else
+        Rails.logger.debug token_response.body
```

Comment:
This is kind of temporary; we've never seen what such an error response might look like and would like to "capture" one for testing purposes.